src/server/editorServices.ts

```diff
@@ -2722,8 +2776,20 @@ namespace ts.server {
                         else {
                             // reload from the disk
                             this.reloadConfiguredProject(project, reason);
+                            if (!project.containsScriptInfo(info) && project.isSolution()) {
+                                forEachResolvedProjectReferenceProject(
+                                    project,
+                                    child => {
+                                        if (!updatedProjects.has(child.canonicalConfigFilePath)) {
+                                            updatedProjects.set(child.canonicalConfigFilePath, true);
+                                            this.reloadConfiguredProject(child, reason);

```

Comment:
I'm still trying to understand the implementation, so this might be a silly question... but here, it seems for a given file `f`, we start at a project and reload it. Then, if it's a solution project, we start looking for the project it's actually in, reloading projects as we go. Then, once we find it, we reload that project and stop looking. Why do we reload the projects it's not in before we find it, but not reload any other projects?

Reply:
This is method that reloads the projects for open files (eg because user invoked that operation in vscode) 
What this does is, it finds config file for each open file.. which if turns out to be solution, we need to reload all projects till we find the project that open file belongs to.. (just like when we do for open file as part of this change)